src/Microsoft.Azure.WebJobs.ServiceBus/MessagingExceptionHandler.cs

```diff
@@ -0,0 +1,138 @@
+            exceptionHandler.Subscribe();
+            return exceptionHandler;
+        }
+
+        public static MessagingExceptionHandler Subscribe(OnMessageOptions options, TraceWriter traceWriter, ILoggerFactory loggerFactory = null)
+        {
+            var exceptionHandler = new ServiceBusExceptionHandler(options, traceWriter, loggerFactory);
+            exceptionHandler.Subscribe();
+            return exceptionHandler;
+        }
+
+        public abstract void Subscribe();
+
+        public abstract void Unsubscribe();
+
+        protected void Handle(object sender, ExceptionReceivedEventArgs e)
+        {
+            LogExceptionReceivedEvent(e);
+        }
+
+        internal void LogExceptionReceivedEvent(ExceptionReceivedEventArgs e)
+        {
+            try
+            {
+                var logger = _loggerFactory?.CreateLogger(LogCategories.Executor);
+                string message = $"{_source} error (Action={e.Action}) : {e.Exception.ToString()}";
+
+                var logLevel = GetLogLevel(e.Exception);
+                logger?.Log(logLevel, 0, message, e.Exception, (s, ex) => message);
+
+                var traceEvent = new TraceEvent(logLevel.ToTraceLevel(), message, null, e.Exception);
+                _traceWriter.Trace(traceEvent);
+            }
+            catch
+            {
+                // best effort logging
+            }
+        }
+
+        protected virtual LogLevel GetLogLevel(Exception ex)
+        {
+            var mex = ex as MessagingException;
+            if (!(ex is OperationCanceledException) && (mex == null || !mex.IsTransient))
+            {
+                // any non-transient exceptions or unknown exception types
+                // we want to log as errors
+                return LogLevel.Error;
+            }
+            else
+            {
+                // transient messaging errors we log as verbose so we have a record
+                // of them, but we don't treat them as actual errors
+                return LogLevel.Information;
+            }
+        }
+
+        private class EventHubExceptionHandler : MessagingExceptionHandler
+        {
+            private readonly EventProcessorOptions _options;
+
+            public EventHubExceptionHandler(EventProcessorOptions options, TraceWriter traceWriter, ILoggerFactory loggerFactory = null) : base("EventProcessorHost", traceWriter, loggerFactory)
+            {
+                _options = options;
+            }
+
+            public override void Subscribe()
+            {
+                _options.ExceptionReceived += Handle;
+            }
+
+            public override void Unsubscribe()
+            {
+                _options.ExceptionReceived -= Handle;
+            }
+
+            protected override LogLevel GetLogLevel(Exception ex)
+            {
+                if (ex is ReceiverDisconnectedException ||
```

Comment:
This is the key change for the EventHub issue fix - overriding the base level determination to make these Info level rather than errors as they are now (generating CRIs)